File: src/test/java/com/ject/studytrip/studylog/helper/StudyLogTestHelper.java
package com.ject.studytrip.studylog.helper;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.studylog.domain.model.StudyLog;
import com.ject.studytrip.studylog.domain.repository.StudyLogRepository;
import com.ject.studytrip.studylog.fixture.StudyLogFixture;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class StudyLogTestHelper {
    @Autowired private StudyLogRepository studyLogRepository;

    public StudyLog saveStudyLog(Member member, DailyGoal dailyGoal) {
        StudyLog studyLog = StudyLogFixture.createStudyLog(member, dailyGoal);
        return studyLogRepository.save(studyLog);
    }
}

File: src/test/kotlin/com/ject/studytrip/studylog/helper/StudyLogTestHelper.kt
package com.ject.studytrip.studylog.helper

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.studylog.domain.model.StudyLog
import com.ject.studytrip.studylog.domain.repository.StudyLogRepository
import com.ject.studytrip.studylog.fixture.StudyLogFixture
import com.ject.studytrip.trip.domain.model.DailyGoal
import org.springframework.stereotype.Component

@Component
class StudyLogTestHelper(
    private val studyLogRepository: StudyLogRepository,
) {
    fun saveStudyLog(
        member: Member,
        dailyGoal: DailyGoal,
    ): StudyLog = studyLogRepository.save(StudyLogFixture(member, dailyGoal).create())

    fun saveDeletedStudyLog(
        member: Member,
        dailyGoal: DailyGoal,
    ): StudyLog = studyLogRepository.save(StudyLogFixture(member, dailyGoal).create().also { it.updateDeletedAt() })
}
